device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -429,4 +361,43 @@ public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeC
         //Codes_SRS_DEVICE_IO_34_020: [This function shall register the callback with the transport.]
         this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext);
     }
+
+    @Override
+    public void onTransportConnectionClosed()
+    {
+        if (taskScheduler != null)
+        {
+            this.taskScheduler.shutdown();
+            this.taskScheduler = null;
+        }
+    }
+
+    @Override
+    public void onTransportConnectionOpened()
+    {
+        if (this.sendTask == null)
+        {
+            this.sendTask = new IotHubSendTask(this.transport);
+        }
+
+        if (this.receiveTask == null)
+        {
+            this.receiveTask = new IotHubReceiveTask(this.transport);
+        }
+
+        if (this.taskScheduler == null)
+        {
+            this.taskScheduler = Executors.newScheduledThreadPool(2);
+
```

Comment:
Executors.newScheduledThreadPool(2); PSF please

Reply:
Sure, I'll make the 2 psf